modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaSttpClientCodegen.java

```diff
@@ -143,6 +143,8 @@ public ScalaSttpClientCodegen() {
         typeMapping.put("binary", "File");
         typeMapping.put("number", "Double");
         typeMapping.put("decimal", "BigDecimal");
+        typeMapping.put("ByteArray", "Array[Byte]");
+        typeMapping.put("ArrayByte", "Array[Byte]");
```

Comment:
why do we need both "ByteArray" and "ArrayByte"? what are the cases that'd make the generator come up with those?

Reply:
Internally DefaultCodegen::sanitizeName(String name) converts "Array[Byte]" to "ArrayByte", that's why need to add "ArrayByte" too.

Reply:
Can you provide a minimal spec for the `ArrayByte` use case? Looks like a custom format not defined in the official openapi spec.

Reply:
@wing328 I was out of town, sorry for the late reply.
`ArrayBytes` use case does not come from openapi spec,  it is because [sanitizeName](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L5436-L5438) converts `Array[Byte]` to `ArrayByte`, so that [getSchemaType(schema)](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L2246) returns `ArrayByte`

Reply:
is it possible to reproduce the issue with openapi spec? it might be related to AbstractScalaCodegen sanitize methods. additional mappings seems as workaround

Reply:
Yeah, additional mappings seems as workaround. Here is a simple openapi spec to reproduce the [issue #10150](https://github.com/OpenAPITools/openapi-generator/issues/10150)
```yaml
openapi: 3.0.3
info:
  title: array item is enum
  version: '1.0'
paths:
  /state:
    get:
      operationId: getUserReport
      properties:
        traces:
          items:
          $ref: '#/components/schemas/TraceWithScreenshotData'
          type: array
      responses:
        '200':
          description: get Users
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Users'
components:
  schemas:
    Users:
      properties:
        search:
          enum:
          - first_name
          - last_name
          - email
          - full_name
          type: string
      type: object
    TraceWithScreenshotData:
      properties:
        screenshot:
          description: |-
            The screenshot image encoded using base64.
            A screenshot should start with a data uri scheme 'data:image/png;base64'.
          format: byte
          type: string
      type: object
```

Reply:
@wing328 @chameleon82 can you review this PR when you get an chance? Thanks

Reply:
@bgong-mdsol I want to include this enhancement but I still couldn't figure out where the `ArrayByte` comes from
```
        typeMapping.put("ArrayByte", "Array[Byte]");
```

If you remove this one, I'm happy to merge this PR.

For `ArrayByte`, I don't think the above example spec illustrates that.

Reply:
@wing328 
Thanks to review my PR. As @chameleon82 mentioned above, this additional mapping is necessary as workaround since DefaultCodegen sanitizeName method converts `Array[Byte]` to `ArrayByte`. It seems that [Scala Http Client ](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaHttpClientCodegen.java#L132) does the same additional mapping too.

Reply:
@wing328  I just simplified the openapi spec above. 
Without the fixes, the generator generates the model class `TraceWithScreenshotData` as 
```
case class TraceWithScreenshotData(
  screenshot: Option[ByteArray] = None    // not found: type ByteArray
)
```
With `typeMapping.put("ByteArray", "Array[Byte]")` only, the generator generates the model class `TraceWithScreenshotData` as 
```
case class TraceWithScreenshotData(
  screenshot: Option[ArrayByte] = None    // not found: type ArrayByte
)
```
With the both fixes, the generator generates the model class TraceWithScreenshotData as expected
```
case class TraceWithScreenshotData(
  screenshot: Option[Array[Byte]] = None 
)
```

Reply:
Thanks for the information. I'm able to repeat the issue and confirmed the following line is needed:
```
        typeMapping.put("ArrayByte", "Array[Byte]");
```

I'll merge it first and may dig into it later to see if there's a way to avoid it.